### DISCUSSION TITLE
[macOS] Distribute a proper app bundle

### DISCUSSION THREAD

**Discussion by deseven** [cdgriffith/FastFlix#587] [Ideas]
While macOS supports just straight up executing binaries, the proper way for apps with GUI is to use app bundles. An app bundle is basically just a directory ending with `.app` that contains a pre-determined structure and a plist file that defines what this app is and how to run it. The OS automatically process such "folders" so they act like an executable instead.

Advantages of using the app bundle in case of FastFlix:
 - app icon will actually be shown everywhere in the system
 - app will be indexed and searchable through Spotlight or other ways used to quickly launch an app
 - it can be pinned in the dock
 - it won't open a terminal window when running (it's still possible to run it in terminal for debugging purposes of course)
 - this is what macOS users actually expect

The whole process is pretty simple and only requires some very basic file manipulations, in addition to converting ICO to ICNS and writing a simple PLIST config.

The resulting structure should look like this:
![](https://d7.wtf/s/SCR-20240720-nfvk.png)

The PLIST should be pretty self-explanatory:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleDevelopmentRegion</key>
	<string>English</string>
	<key>CFBundleExecutable</key>
	<string>FastFlix</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleIconFile</key>
	<string>icon.icns</string>
	<key>CFBundleIdentifier</key>
	<string>com.github.cdgriffith.FastFlix</string>
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>CFBundleSignature</key>
	<string>PURE</string>
	<key>CFBundleVersion</key>
	<string>5.7.4</string>
	<key>CFBundleName</key>
	<string>FastFlix</string>
	<key>CSResourcesFileMapped</key>
	<true/>
	<key>NSHighResolutionCapable</key>
	<true/>
	<key>NSDisablePersistence</key>
	<true/>
	<key>LSMinimumSystemVersion</key>
	<string>12.0</string>
</dict>
</plist>
```

~~Here's the latest release (5.7.4) packed into an app bundle as a **proof of concept**:~~
~~REMOVED FOR SAFETY~~

**Comment by cdgriffith** [comment]
Had no idea it was that simple, will play around with it this weekend, thanks for the info!

**Comment by cdgriffith** [comment]
Added test versions with `appbundle` in name to https://github.com/cdgriffith/FastFlix/releases/tag/5.8.0b0 if @deseven or others can test!

**Comment by deseven** [comment]
@cdgriffith sorry, for some reason GH decided to not notify me about this topic, even with direct mentions. I've tested the latest 5.8.0 release and the app bundle worked great! Thanks.
